### DISCUSSION TITLE
Symbols visibility

### DISCUSSION THREAD

**Discussion by EduPonz** [eProsima/Fast-CDR#149] [Ideas]
This discussion's intent is to define a better approach from symbols' visibility. It has sprouted from #88 and #89 by @roehling.
Important points to clarify would be:

1. Fix the problem of exporting the library exception classes, which right now are not exported per se, but in a member function by member function basis instead, since they inherit from `std::exception` which is not exported.
   For this case, #88 proposed simply disabling the MSVC warnings, but does not seem like a solution to the problem either, so it was discarded.
2. Whether to set the `-fvisibility=hidden` and ` -fvisibility-inlines-hidden` flags in GCC so that only the specified symbols are exported (similar to MSVC approach)

**Comment by MiguelCompany** [comment]
We've been discussing this, and think that in order to properly fix visibility of exception classes in MSVC, we would need a breaking change where our base exception does not inherit from `std::exception`, adding it by composition instead
